### PR TITLE
MINOR: [Docs][Java] Update status documentation for Java's CSV read support (as per ARROW-17786)

### PR DESCRIPTION
**Pull request by igor-suhorukov**
Update status documentation: CSV read in Java from #14182

### CODE REVIEW

File: docs/source/status.rst [hunk 1]
```diff
@@ -331,7 +331,7 @@ Third-Party Data Formats
 +=============================+=========+=========+=======+============+=======+=======+=======+
 | Avro                        |         | R       |       |            |       |       |       |
 +-----------------------------+---------+---------+-------+------------+-------+-------+-------+
-| CSV                         | R/W     |         | R/W   |            |       | R/W   | R/W   |
+| CSV                         | R/W     | R       | R/W   |            |       | R/W   | R/W   |
```
Comment (lidavidm) [inline docs/source/status.rst:334]: We should add the caveat that it's through JNI (like for ORC below).

**Review by lidavidm — COMMENTED** [review]
Thanks, just one comment.

File: docs/source/status.rst (hunk 1 above)
Comment (igor-suhorukov) [reply docs/source/status.rst]: Hello @lidavidm Thank you for review! Done

**Review by lidavidm — APPROVED** [review]
Thank you!

**Pull request merged**